src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#if NETCOREAPP
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Build.Locator
+{
+    internal static class DotNetSdkLocationHelper
+    {
+        private static readonly Regex DotNetBasePathRegex = new Regex("Base Path:(.*)$", RegexOptions.Multiline);
+        private static readonly Regex VersionRegex = new Regex(@"^(\d+).(\d+).(\d+)", RegexOptions.Multiline);
```

Comment:
```suggestion
        private static readonly Regex VersionRegex = new Regex(@"^(\d+)\.(\d+)\.(\d+)", RegexOptions.Multiline);
```

I lucked into an extremely confusing error mode! The SDK I had installed happened to fit the earlier pattern, since `.` could match the non-decimal hex digits:

```
91e160c7f04c8f966895d441714b6dd6697a2f94
2.1.504
win-x64
```

gave me version `91.160.7`.

Reply:
haha thanks. It's been a while since I regex'd.